Primo.js

```diff
@@ -369,9 +384,51 @@ function importPNX(text) {
 		item.pages = endPage;
 	}
 	
-	// does callNumber get stored anywhere else in the xml?
-	item.callNumber = ZU.xpathText(doc, '//enrichment/classificationlcc');
+	//these are actual local full text links (e.g. to google-scanned books)
+	//e.g http://solo.bodleian.ox.ac.uk/OXVU1:LSCOP_OX:oxfaleph013370702
+	var URL = ZU.xpathText(doc, '//links/linktorsrc');
+	if (URL && URL.search(/\$\$U.+\$\$/) != -1) {
+		item.url = URL.match(/\$\$U(.+?)\$\$/)[1];
+	}
+
+	//add finding aids as links
+	var findingAid = ZU.xpathText(doc, '//links/linktofa');
+	if (findingAid && findingAid.search(/\$\$U.+\$\$/) != -1) {
+		item.attachments.push({url: findingAid.match(/\$\$U(.+?)\$\$/)[1], title: "Finding Aid", snapshot: false});
+	}
+	// get the best call Number; sequence recommended by Harvard University Library
+	var callNumber = ZU.xpath(doc, '//browse/callnumber');
+	var callArray = [];
+	for (var i = 0; i<callNumber.length; i++) {
+		if (callNumber[i].textContent.search(/\$\$D.+\$/) != -1) {
+			callArray.push(callNumber[i].textContent.match(/\$\$D(.+?)\$/)[1]);
+		}
+	}
+	if (!callArray.length) {
+		callNumber = ZU.xpath(doc, '//display/availlibrary');
+		for (var i = 0; i<callNumber.length; i++) {
+			if (callNumber[i].textContent.search(/\$\$2.+\$/) != -1) {
+				callArray.push(callNumber[i].textContent.match(/\$\$2\(?(.+?)(?:\s*\))?\$/)[1]);
```

Comment:
this looks to me like I should be removing duplicates in the array before converting it to a string. Will do that.